### PR TITLE
Disabling suggestions on Android if readonly mode is set.

### LINKED ISSUE #2136: Android Read-only Multi-line text input displays suggestions (Spell Checking)
### Describe the bug

When using the "MultilineTextInput" with the "readonly" parameter set to "True" on Android, it will suggest spelling improvements based on dictionary candidates. This behavior results in underlined text on android devices when using a language other than the keyboard language or with unknown words in general.

### Steps to reproduce

1. Use MultilineTextInput in readonly with a set text
2. Compile for android and set the KeyBoard language to a different one than the text.
3. The suggestions should appear, underlining the Text.

### Expected behavior

Suggestions do not show up in MultilineTextInput if set to readonly.

### Screenshots

_No response_

### Environment

- Operating System: Fedora 38 Workstation / Lineage OS 20.0
- Python version: 3.11.5
- Software versions:
  - Briefcase: 0.3.15
  - Toga: 0.3.1 

[tool.briefcase.app.trinker.android]
requires = [
    #"toga-android~=0.3.1"
    'git+https://github.com/beeware/toga.git#subdirectory=core',
    'git+https://github.com/beeware/toga.git#subdirectory=android',
]

### Logs

```

```

### Additional context

I did some research on the topic and while I'm not really an Android dev, I think I found the problem. In "android/src/toga_android/widgets/textinput.py" we need to set the flag [TYPE_TEXT_FLAG_NO_SUGGESTIONS](https://developer.android.com/reference/android/text/InputType#TYPE_TEXT_FLAG_NO_SUGGESTIONS) if "readonly" is true.

As said before, I'm not really an Android dev and am not sure how this would be implemented.

### PR DESCRIPTION
**Pull request by Aarontheissueguy**
Set "TYPE_TEXT_FLAG_NO_SUGGESTIONS" as "InputType" if "readonly" is True and testing it in the backend probe

This ensures that no spelling suggestions are made if a text is set to readonly. 

Fixes #2136

### CODE REVIEW

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Looks good, and works well in my testing.

There's a couple of things standing in the way of me merging this:
1. When you submit a PR that closes a ticket, you need to include the text `Fixes #xxx` in the PR description; this is the trigger to Github to close the issue when the PR is merged. I've made this modification on your behalf.
2. Toga's CI requires that every PR include [a changenote](https://toga.readthedocs.io/en/stable/how-to/contribute-code.html#add-change-information-for-release-notes) - this is why CI is currently failing. In this case, we need a file named `changes/2136.bugfix.rst` , with a single line of content that reads something like "Multiline text inputs no longer show spelling suggestions when in readonly mode."
3. When you submit a PR, it's really helpful if you submit the PR from a *branch* of your fork, rather than the main branch. The missing changenote is a fairly minor thing, so ordinarily I would add it myself as part of the review process - but because you've submitted from your `main` branch, I can't do that: maintainers can edit PRs submitted from a *branch*, but not PRs submitted from your main. Submitting from a branch also makes *your* life easier in the long run, because it allows you to have multiple PRs in flight at the same time (after all, you can only have 1 main) - but once your PR is merged, it also means the reconciliation process is easier, because your fork's mainline can stay a "clean" copy of  the upstream repository, rather than something you need to merge or rebase.

There's no need to re-work this PR to be from a different branch; if you add the changenote, I'll be happy to merge it. But it's worth keeping in mind for next time.

**Comment by Aarontheissueguy** [comment]
Thanks for the tips, I will keep them in mind the next time I contribute. I added the changenote as requested.

**Comment by freakboy3742** [comment]
Getting closer - now the PR is hitting a pre-commit issue. We apply black formatting to format our code, and we check that every commit has had black applied. If you follow the [contributors guide](https://toga.readthedocs.io/en/stable/how-to/contribute-code.html#set-up-your-development-environment) you can install pre-commit locally, and have these fixes applied automatically when you commit code. In this case, you'll need to run `pre-commit run --all` over your code to apply the fixes to previous commits.

**Comment by freakboy3742** [comment]
Ok - now we're at a "real" problem, not just a cosmetic one: The test you've added is failing in CI.

**Comment by Aarontheissueguy** [comment]
hmm... Il look into it. Not entirely sure what causes this though.

File: android/tests_backend/widgets/textinput.py
Comment (freakboy3742) [inline android/tests_backend/widgets/textinput.py:50]: The `readonly` property is not *asserting* that the widget is readonly - it is returning the current readonly status (hence the `return not focusable`). This test is asserting that TYPE_TEXT_FLAG_NO_SUGGESTIONS is *always* set. 

```suggestion
        if input_type & InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS:
            # TYPE_TEXT_FLAG_NO_SUGGESTIONS is set
            if focusable:
                raise ValueError(
                    "TYPE_TEXT_FLAG_NO_SUGGESTIONS is not set on the input."
                )
        else:
            # TYPE_TEXT_FLAG_NO_SUGGESTIONS is not set
            if not focusable:
                raise ValueError(
                    "TYPE_TEXT_FLAG_NO_SUGGESTIONS has been set when the input is readonly."
                )
```

**Comment by Aarontheissueguy** [comment]
Thats weird, we didn't tamper with anything iOS related, did we?

**Comment by freakboy3742** [comment]
The good news (for you) is that the iOS failure has nothing to do with you. I landed an unrelated change to Briefcase this morning, and that is causing problems with Toga's CI.

**Review by freakboy3742 — APPROVED** [review]
The CI problem has now been fixed - so this is good to land! 

Thanks for contribution!

**Pull request merged**
